docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,324 @@
+ * Name: mpu_userintsram
+ *
+ * Description:
+ *   Configure a region as user internal SRAM
+ *
+ ****************************************************************************/
+static inline void mpu_userintsram(uint32_t region, uintptr_t base, size_t size)
+{
+.
+.
+.
+}
+```
+
+- In case of App-Binary Separation, below function is used to configure the user application SRAM MPU settings.
+
+```c
+
+/****************************************************************************
+ * Name: mpu_get_register_value
+ *
+ * Description:
+ *   Configure the user application SRAM mpu settings into the tcb variables
+ *
+ * Params:
+ *  regs     : pointer to array in which to store the configured values
+ *             If regs is NULL, then configure the mpu registers directly
+ *  region   : number of the region to be configured
+ *  base     : start address for the region
+ *  size     : size of the region in bytes
+ *  readonly : true indicates a readonly region
+ *  execute  : true indicates that the region has execute permission
+ ****************************************************************************/
+ 
+void mpu_get_register_value(uint32_t \*regs, uint32_t region, uintptr_t base, size_t size, uint8_t readonly, uint8_t execute)
+{
+.
+.
+.
+}
+```
+
+- Mapping of MPU regions to their respective memory segments has been done statically using below enumeration (in os/include/tinyara/mpu.h)
+
+```c
+enum MPU_REG_NUM {
+#ifdef CONFIG_APP_BINARY_SEPARATION
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	MPU_REG_NUM_COM_LIB_TXT,	/* Common Binary */
+	MPU_REG_NUM_COM_LIB_RO,
+	MPU_REG_NUM_COM_LIB_DATA,
+#else
+	MPU_REG_NUM_COM_LIB,
+#endif
+#endif
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	MPU_REG_NUM_APP_TXT,		/* Apps */
+	MPU_REG_NUM_APP_RO,
+	MPU_REG_NUM_APP_DATA,
+#else
+	MPU_REG_NUM_APP,
+#endif
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+	MPU_REG_NUM_STK,		/* Stack */
+#endif
+#else
+	MPU_REG_NUM_UFLASH,
+	MPU_REG_NUM_URAM,
+#endif
+	MPU_REG_NUM_MAX
+};
+```
+
+## How to Control MPU regions in TizenRT
+### Add a new MPU region
+1. Add your MPU region in enum MPU_REG_NUM in mpu.h file.
+2. Call function mpu_get_register_value to configure your MPU region with appropriate arguments.
+	- If (uint32_t \*)regs argument is passed as NULL, MPU region settings will be directly written into MPU registers
+	- If instead a variable (like uint32_t var[3]) is passed as argument, then MPU register values are saved in this variable and can be used at appropriate place for setting MPU region (ex. during context switch).
+
+### Delete a MPU region
+1. Remove MPU region from enum MPU_REG_NUM in mpu.h file.
+2. Remove calls made to mpu_get_register_value using this MPU region as the (uint32_t) region argument.
+
+### Modify a MPU region
+1. For overlapping memory address range between 2 MPU regions, modify their relative position in enum MPU_REG_NUM as per below rule:
```

Comment:
Since only one point is present, we can remove the numbering. 
Also, its better if we can reframe as follows:

When modifying the mpu regions, care must be taken for regions with overlapping address range, by following the below guidelines.

Reply:
done